pkg/distributor/errors.go

```diff
@@ -196,6 +196,22 @@ func (e ingesterPushError) errorCause() mimirpb.ErrorCause {
 // Ensure that ingesterPushError implements distributorError.
 var _ distributorError = ingesterPushError{}
 
+type circuitBreakerOpenError struct {
+	client.ErrCircuitBreakerOpen
```

Comment:
Why use anonymous (embedded) field here? It will make `circuitBreakerOpenError` to inherit all methods from `client.ErrCircuitBreakerOpen`. Unless there's a specific reason, I'd prefer to avoid that.

Reply:
I thought a bit more and I decided to explicitly add the `RemainingDelay()` function to `circuitBreakerOperError`, so that its interface doesn't depend on the interface of `client.ErrCircuitBreakerOpen`.